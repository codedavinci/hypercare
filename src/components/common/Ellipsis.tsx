type EllipsisProps = {
  text: string;
};

const Ellipsis = ({ text }: EllipsisProps) => {
  return <div className="line-clamp-3">{text}</div>;
};

Ellipsis.displayName = "Ellipsis";

export default Ellipsis;
