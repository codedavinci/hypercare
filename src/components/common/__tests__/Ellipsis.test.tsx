import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Ellipsis from "../Ellipsis";

describe("Ellipsis Component", () => {
  const sampleText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";

  it("should render correctly and match snapshot", () => {
    const { asFragment } = render(<Ellipsis text={sampleText} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the provided text", () => {
    render(<Ellipsis text={sampleText} />);
    expect(screen.getByText(sampleText)).toBeInTheDocument();
  });

  it("should apply the correct class for line clamping", () => {
    render(<Ellipsis text={sampleText} />);
    const ellipsisElement = screen.getByText(sampleText);
    expect(ellipsisElement).toHaveClass("line-clamp-3");
  });
});
