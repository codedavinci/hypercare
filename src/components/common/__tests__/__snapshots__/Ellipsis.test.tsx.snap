// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ellipsis Component should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="line-clamp-3"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </div>
</DocumentFragment>
`;
